crates/oxc_linter/src/ast_util.rs

```diff
@@ -202,3 +204,42 @@ impl<'a> IsPrivate for PropertyDefinition<'a> {
         self.accessibility.map_or(false, |accessibility| accessibility == TSAccessibility::Private)
     }
 }
+
+pub fn get_enclosing_function<'a, 'b>(
+    node: &'b AstNode<'a>,
+    ctx: &'b LintContext<'a>,
+) -> Option<&'b AstNode<'a>> {
+    let mut current_node = node;
+    loop {
+        if matches!(current_node.get().kind(), AstKind::Root) {
+            return None;
+        }
+        if matches!(current_node.get().kind(), AstKind::Function(_) | AstKind::ArrowExpression(_)) {
+            return Some(current_node);
+        }
+        current_node = ctx.parent_node(current_node).unwrap();
+    }
+}
+
+pub fn is_nth_argument<'a>(call: &CallExpression<'a>, arg: &Argument<'a>, n: usize) -> bool {
+    let nth = &call.arguments[n];
+    nth.span() == arg.span()
+}
```

Comment:
This is really useful, can you add a doc comment?

Reply:
Done.